subsys/bluetooth/audio/mcc.c

```diff
@@ -1472,8 +1685,9 @@ int bt_mcc_init(struct bt_mcc_cb *cb)
  * The order of discovery is follows:
  * 1: Discover GMCS primary service (started here)
  * 2: Discover characteristics of GMCS
+ * 3: Subscribe to characteristics of GMCS
  * 3: Discover OTS service included in GMCS
- * 4: Discover characteristics of OTS
+ * 4: Discover characteristics of OTS and subscribe to them
```

Comment:
Can't help but to wonder that if we do these steps in this order, that the OTS characteristic discovery function should be split from the MCS discovery function, instead of guarding the individual characteristics with `#ifdef CONFIG_BT_MCC_OTS`. 

Not needed to be modified by this PR of course.

Reply:
Ah - this is the confusing part of MCS.
The OTS characteristic discovery function is split out as a separate function - `discover_otc_char_func()`
The individual characteristics that are guarded are MCS characteristics, but they are dependent upon there being an MCS.

Reply:
Ah, right. So it's not as trivial to "fix".